src/cmake-tools.ts

```diff
@@ -599,17 +606,64 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
                 progress.report({message: localize('configuring.project', 'Configuring project')});
                 let retc: number;
                 switch (type) {
-                case ConfigureType.Normal:
-                  retc = await drv.configure(extra_args, consumer);
-                  break;
-                case ConfigureType.Clean:
-                  retc = await drv.cleanConfigure(extra_args, consumer);
-                  break;
-                default:
-                  rollbar.error(localize('unexpected.configure.type', 'Unexpected configure type'), {type});
-                  retc = await this.configure(extra_args, ConfigureType.Normal);
-                  break;
+                  case ConfigureType.Normal:
+                      retc = await drv.configure(extra_args, consumer);
+                    break;
+                  case ConfigureType.Clean:
+                      retc = await drv.cleanConfigure(extra_args, consumer);
+                    break;
+                  default:
+                      rollbar.error(localize('unexpected.configure.type', 'Unexpected configure type'), {type});
+                      retc = await this.configure(extra_args, ConfigureType.Normal);
+                    break;
+                }
+                if (retc === 0) {
+                  await this._refreshCompileDatabase(drv.expansionOptions);
+                }
+                await this._ctestController.reloadTests(drv);
+                this._onReconfiguredEmitter.fire();
+                return retc;
+              } finally {
+                progress.report({message: localize('finishing.configure', 'Finishing configure')});
+                prog_sub.dispose();
+              }
+            } else {
+              progress.report({message: localize('configure.failed', 'Failed to configure project')});
+              return -1;
+            }
+          });
+        },
+    );
+  }
+
+  /**
+   * Configure without cmake settings
+   */
+  rawConfigure(): Thenable<number> {
```

Comment:
I think this could be factored better.  I don't see any significant difference between this function and `configure` besides that this one passes in `[]` and an extra `true` to `drv.configure`.  We should avoid copying large portions of the code around because it is cumbersome and error-prone to keep both functions up to date.

Instead, you should create a new `ConfigureType` (maybe `ConfigureType.Cache`) and add a case label to the existing `configure` function.  Then rename `rawConfigure` to `cacheConfigure` and just call `configure([], ConfigureType.Cache)` here.